src/lesson20/js/userTable.js

```diff
@@ -0,0 +1,101 @@
+import { renderCircle } from "./modules/renderCircle.js";
+import { removeCircle } from "./modules/removeCircle.js";
+import { displayInfo } from "./modules/displayInfo.js";
+
+const url = "https://mocki.io/v1/4689014e-270d-46d3-889f-72647a3d830a";
+
+function renderStatus(response) {
+  const errorMessage = document.createElement("p");
+  errorMessage.textContent = `${response.status}:${response.statusText}`;
+  document.body.appendChild(errorMessage);
+}
+
+const fragment = document.createDocumentFragment();
+
+function makeHeaderRow(dataSet) {
+  const theadTag = document.createElement("thead");
+  dataSet.forEach((data) => {
+    const row = document.createElement("tr");
+    Object.keys(data).forEach((key) => {
+      const thTag = document.createElement("th");
+      thTag.textContent = data[key];
+      thTag.classList.add("th");
+      row.appendChild(thTag);
+    });
+    fragment.appendChild(theadTag).appendChild(row);
+  });
+  return fragment;
+}
```

Comment:
IMO
`makeHeaderData` でオブジェクトにする必要はない気がしました！
`makeTable`の中で`const headerData = Object.keys(dataSet.data[0]);`として引数に渡してあげれば
下記の様に実装することも可能です
```suggestion
function makeHeaderRow(dataSet) {
  const theadTag = document.createElement("thead");
  const row = document.createElement("tr");
  dataSet.forEach((data) => {
      const thTag = document.createElement("th");
      thTag.textContent = data.toUpperCase();
      thTag.classList.add("th");
      row.appendChild(thTag);
    });
    fragment.appendChild(theadTag).appendChild(row);
  return fragment;
}
```

Reply:
> IMO `makeHeaderData` でオブジェクトにする必要はない気がしました！ `makeTable`の中で`const headerData = Object.keys(dataSet.data[0]);`として`headerData`を引数に渡してあげれば 下記の様に実装することも可能です

Thank you sae-san :)
Your advice meke my code simple :)
9adad65917f9602b1629959cbc40c1bf4ef40216
a4af32abf6a3fd81b59ead93a56ff560950c2889